### DISCUSSION TITLE
Delay on response before moving to next trial

### DISCUSSION THREAD

**Discussion by rayoverweij** [jspsych/jsPsych#1909] [Q&A] [answered]
Hi there,

It feels like I'm overlooking something simple here but I figured I'd just ask. I'm building a decision-making task where the participant has to choose between two images in each trial, using `html-button-response` with images in the buttons. Upon selecting one of the images, the selected image needs to play an animation before the next trial is started.

The animation is completely controlled in CSS - I just need a 1s delay from the moment the button is pressed until `on_finish` triggers. How should I go about this?

Cheers!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @rayoverweij, this should be pretty straightforward if you don't mind editing the plugin file. If I understand correctly, then I think what you want to do is insert a delay between the response and the end of the trial. You can do that by editing the `after_response` function in the plugin. Specifically, this bit of code checks to see whether the trial should end after a response, and if so, then it immediately ends the trial:
https://github.com/jspsych/jsPsych/blob/7b16a1d39ab38b3a01c33c9d0a2fec06524e81ee/plugins/jspsych-html-button-response.js#L146-L148

So you can insert a delay between the response and the end of the trial here using a `setTimeout` timer. I recommend using the [jsPsych.pluginAPI.setTimeout function](https://www.jspsych.org/core_library/jspsych-pluginAPI/#jspsychpluginapisettimeout) for this, because this will mean that the timer will be cleared at the end of the trial (which is good practice as it prevents a timer from firing during a later trial, which would cause all kinds of weird bugs). Here's the code you can use:
```js
if (trial.response_ends_trial) {
  // after the participant responds, wait 1000 ms, then call the end_trial function
  jsPsych.pluginAPI.setTimeout(function() {
    end_trial();
  }, 1000); 
}
```
Does this work for you?

**Reply by rayoverweij** [reply]
That worked perfectly, thanks a lot!
